include/aws/io/channel_bootstrap.h

```diff
@@ -114,12 +114,19 @@ typedef void(aws_server_bootstrap_on_accept_channel_setup_fn)(
  * Note: this function is only invoked if the channel was successfully setup,
  * e.g. aws_server_bootstrap_on_accept_channel_setup_fn() was invoked without an error code.
  */
-typedef void(aws_server_bootsrap_on_accept_channel_shutdown_fn)(
+typedef void(aws_server_bootstrap_on_accept_channel_shutdown_fn)(
     struct aws_server_bootstrap *bootstrap,
     int error_code,
     struct aws_channel *channel,
     void *user_data);
 
+/**
+ * Once the server listener socket is finished destroying, all the existing connection is closed, this fuction will be
+ * invoked.
+ */
+typedef void(
+    aws_server_bootstrap_on_server_listener_destroy_fn)(struct aws_server_bootstrap *bootstrap, void *user_data);
```

Comment:
modify the comments of aws_server_bootstrap_new_socket_listener(), aws_server_bootstrap_new_tls_socket_listener() and aws_server_bootstrap_destroy_socket_listener() so they mention this function. Something like:

`destroy_callback` will be invoked after the server socket listener is destroyed, and all associated connections and channels have finished shutting down.